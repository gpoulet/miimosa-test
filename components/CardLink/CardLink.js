import { string } from 'prop-types'
import Link from 'next/link'
import Button from '@material-ui/core/Button'

function CardLink({ href, label }) {
  return (
    <Link href={href}>
      <Button size="small" color="primary">
        <a>{label}</a>
      </Button>
    </Link>
  )
}

CardLink.propTypes = {
  href: string.isRequired,
  label:string.isRequired
}

export default CardLink