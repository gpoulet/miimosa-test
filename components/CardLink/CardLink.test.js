import React from 'react';
import renderer from 'react-test-renderer';
import CardLink from './CardLink';

it('CardLink renders correctly', () => {
  const tree = renderer
    .create(<CardLink href="/test"label="Lien de test" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});