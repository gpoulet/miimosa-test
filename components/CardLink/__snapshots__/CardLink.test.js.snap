// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardLink renders correctly 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    <a>
      Lien de test
    </a>
  </span>
</button>
`;
